python_modules/dagster/dagster/_core/definitions/selector.py

```diff
@@ -373,31 +373,35 @@ def from_graphql_input(graphql_data):
         )
 
 
-class PartitionsSelector(
-    NamedTuple(
-        "_PartitionsSelector",
-        [("partition_range", PartitionRangeSelector)],
-    )
-):
-    """The information needed to define selection partitions.
-    Using partition_range as property name to avoid shadowing Python 'range' builtin .
-    """
+class PartitionsSelector(NamedTuple):
+    """The information needed to define selection partitions."""
 
-    def __new__(cls, partition_range: PartitionRangeSelector):
-        return super(PartitionsSelector, cls).__new__(
-            cls,
-            partition_range=check.inst_param(partition_range, "range", PartitionRangeSelector),
-        )
+    ranges: Sequence[PartitionRangeSelector]
 
     def to_graphql_input(self):
-        return {
-            "range": self.partition_range.to_graphql_input(),
-        }
+        return {"ranges": [partition_range.to_graphql_input() for partition_range in self.ranges]}
 
     @staticmethod
-    def from_graphql_input(graphql_data):
-        return PartitionsSelector(
-            partition_range=PartitionRangeSelector.from_graphql_input(graphql_data["range"])
+    def from_graphql_input(graphql_data: Mapping[str, Any]) -> "PartitionsSelector":
+        if "ranges" in graphql_data:
+            check.invariant(
```

Comment:
I think this is fine, especially if we think most of the usage is driven by the API vs ad-hoc user manipulation.

If we truly had this endpoint be supported / public, we'd want there to be a special error type (e.g. `InvalidBackfillInputArguments`) or something instead of the `PythonError` we would show.